doc/source/building/windows.rst

```diff
@@ -8,155 +8,268 @@ Building From Source on Windows
-hours but may also fail silently and produce an incorrect binary. Please, if you
-have any issues, `report them`_ so that we can save the next person's time.
 
-While you're waiting on OpenBLAS to finish building, go ahead and install `build tools`_
-from Microsoft, since these take a while to install and you'll need them later.
+Then open this file in a text editor like Notepad++ and paste the following
+content in this empty file:
+
+.. code:: shell
+
+    # Adjust the following to your liking and your MSYS installation status
+    OPENBLAS_ROOT=/c/opt
+    BUILD_BITS=64
+
+    # Print some gcc info that MSYS2 discovered in the path
+    which gcc
+    gcc --version
+
+    # Get into the repository that we cloned
+    cd OpenBLAS
+
+    # Change the following to a specific branch/tag/release you wish
+    # Consult the git manual to learn more about your options
+    git checkout tags/v0.3.7 -b v0.3.7
+
+    # The following two lines clean up in case we make a mistake and need
+    # to run the script again
+    git clean -fxd
+    rm -rf $OPENBLAS_ROOT/$BUILD_BITS
+
+    # Set architecture flags
+    march="x86-64"
+    extra="-fno-asynchronous-unwind-tables"
+    vc_arch="X64"
+    cflags="-O2 -march=$march -mtune=generic $extra"
+    fflags="$cflags -frecursive -ffpe-summary=invalid,zero"
+
+    # Build name for output library from gcc version and OpenBLAS commit.
+    GCC_TAG="gcc_$(gcc -dumpversion | tr .- _)"
+    OPENBLAS_VERSION=$(git describe --tags)
+    # Build OpenBLAS
+    # Variable used in creating output libraries
+    export LIBNAMESUFFIX=${OPENBLAS_VERSION}-${GCC_TAG}
+    make BINARY=$BUILD_BITS DYNAMIC_ARCH=1 USE_THREAD=1 USE_OPENMP=0 \ 
+        NO_WARMUP=1 BUILD_LAPACK_DEPRECATED=1 \ 
+        COMMON_OPT="$cflags" FCOMMON_OPT="$fflags"
+    make install PREFIX=$OPENBLAS_ROOT/$BUILD_BITS
+
+This is the automation script that will make sure the right variables are used
+in the right place. Linux users are very familiar to such scripts but for
+Windows users it might be a bit awkward. You can think of these as ``.bat``
+files. You can change the variables to your situation. After you've created this
+file and you are one directory up the OpenBLAS repo of that, start the OpenBLAS
+build with:
+
+.. code:: shell
+
+    ./build_openblas.sh
 
-After the :code:`build_openblas.sh` script has completed (probably with an error), there
-should be an :code:`openblas.a` file somewhere on your system. If :code:`OPENBLAS_ROOT` was
-set to :code:`C:\\opt`, then you might see a line like this in the MSYS2 terminal:
+Building OpenBLAS is challenging and time-consuming. The build may fail with an
+error after a few hours but may also fail silently and produce an incorrect
+binary. Please, if you have any issues, `report them`_ so that we can save the
+next person's time.
+
+One of the known issues is the following; if you, by any chance, receive the
+following error
```

Comment:
```suggestion
following error:
```
ugh... Sphynx

Reply:
it was the missing whitespace before `shell`